sn_node/src/node/messaging/agreement.rs

```diff
@@ -23,22 +23,26 @@ impl MyNode {
         &mut self,
         proposal: Proposal,
         sig: SectionSig,
-    ) -> Result<Option<Cmd>> {
+    ) -> Result<Vec<Cmd>> {
         debug!("{:?} {:?}", LogMarker::ProposalAgreed, proposal);
+        let mut cmds = Vec::new();
         match proposal {
             Proposal::VoteNodeOffline(node_state) => {
-                Ok(self.handle_offline_agreement(node_state, sig))
+                if let Some(cmd) = self.handle_offline_agreement(node_state, sig) {
+                    cmds.push(cmd);
+                }
```

Comment:
I think you can write this as an extend as well:
```suggestion
                cmds.extend(self.handle_offline_agreement(node_state, sig))
```